### PR TITLE
update readme w/ benefits/goals of s2i

### PR DESCRIPTION
**Pull request by bparees**
/cc @smarterclayton

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton updated.

File: README.md [hunk 1]
```diff
@@ -7,9 +7,20 @@
...
 ready-to-run images by injecting source code into a Docker image and *assembling*
-a new Docker image which incorporates the builder image and built source.  The result is then ready to use
-with `docker run`. `s2i` supports incremental builds which re-use previously downloaded
-dependencies, previously built artifacts, etc.
+a new Docker image for the application.  The result is then ready to use with `docker run`. 
+
+The basic S2I flow is:
+
+1. Start with a ruby builder image which contains ruby, bundler, rake, apache, gcc, etc, all of which are needed to install and run ruby code.  
+1. S2I launches a container using the builder image and injects the provided source code into a working directory within the container.
```
Comment (smarterclayton) [inline README.md]: No need to say `s2i`, just say "Launch a container ... and inject".

File: README.md [hunk 2]
```diff
@@ -7,9 +7,20 @@
...
+1. The container assembles the source code into a runnable structure.
```
Comment (smarterclayton) [inline README.md]: The container then transforms that source code into the appropriate runnable setup - in this case, by moving the source code into a directory where Apache has been preconfigured to look for a Ruby app.

File: README.md [hunk 3]
```diff
@@ -7,9 +7,20 @@
...
+1. S2I commits the new container, setting the default run command for the new image to a script that will launch the assembled application.
```
Comment (smarterclayton) [inline README.md]: Commit the new container and set the image entrypoint to be a script (provided by the builder image) that will start Apache to host the Ruby app.

File: README.md [hunk 4]
```diff
@@ -7,9 +7,20 @@
 
 Source-to-image (`s2i`) is a tool for building reproducible Docker images. `s2i` produces
```
Comment (smarterclayton) [inline README.md]: `Source-to-Image`

File: README.md [hunk 5]
```diff
@@ -7,9 +7,20 @@
...
+1. S2I commits the new container, setting the default run command for the new image to a script that will launch the assembled application.
+
+S2I supports two primary patterns:
+* Produce an application image which includes all development/build tooling used to assemble the application as well as the runtime environment needed to execute the application.
+* Produce an application image which only contains the final runnable application and the runtime environment.
+
+The first pattern is appropriate for dynamic languages which typically need the same tools for building as execution.  The second can be used for compiled languages such as Java when it is undesirable to include build tools such as `javac` and `maven`.  Including build tools in the application image results in a self-contained image which can fully reproduce the application using the exact tools that originally created it.  More sophisticated flows make it possible to separate those two images into one image containing build inputs, build tools, and built artifacts and another image containing the built artifacts (provided as "source") and the runtime environment.
```
Comment (smarterclayton) [inline README.md]: undesirable to include ... `in the runtime image`.

File: README.md [hunk 6]
```diff
@@ -7,9 +7,20 @@
...
+a new Docker image for the application.  The result is then ready to use with `docker run`. 
+
+The basic S2I flow is:
+
+1. Start with a ruby builder image which contains ruby, bundler, rake, apache, gcc, etc, all of which are needed to install and run ruby code.  
+1. S2I launches a container using the builder image and injects the provided source code into a working directory within the container.
+1. The container assembles the source code into a runnable structure.
+1. S2I commits the new container, setting the default run command for the new image to a script that will launch the assembled application.
+
+S2I supports two primary patterns:
```
Comment (smarterclayton) [inline README.md]: There are two core patterns when building images:

File: README.md [hunk 7]
```diff
@@ -7,9 +7,20 @@
...
+* Produce an application image which includes all development/build tooling used to assemble the application as well as the runtime environment needed to execute the application.
+* Produce an application image which only contains the final runnable application and the runtime environment.
+
+The first pattern is appropriate for dynamic languages which typically need the same tools for building as execution.  The second can be used for compiled languages such as Java when it is undesirable to include build tools such as `javac` and `maven`.  Including build tools in the application image results in a self-contained image which can fully reproduce the application using the exact tools that originally created it.  More sophisticated flows make it possible to separate those two images into one image containing build inputs, build tools, and built artifacts and another image containing the built artifacts (provided as "source") and the runtime environment.
 
 Interested in learning more? Read on!
```
Comment (smarterclayton) [inline README.md]: Unnecessary

Comment (smarterclayton) [inline README.md]: Next sentence can be simplified as well - `Try building your own image`

**Comment by bparees** [comment]
[merge]

**Pull request merged**